fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/temp/Type.java

```diff
@@ -240,6 +243,10 @@ public boolean equals(final Object obj) {
                     return false;
                 }
 
+                if (obj == this) {
```

Comment:
ah - duh! :-) sorry for wasting all your time in that meeting about this :-)

Reply:
No worries!